src/main/java/com/puppycrawl/tools/checkstyle/JavadocDetailNodeParser.java

```diff
@@ -125,10 +133,22 @@ public ParseStatus parseJavadocAsDetailNode(DetailAST javadocCommentAst) {
             result.setTree(tree);
         }
         catch (ParseCancellationException | IllegalArgumentException ex) {
-            // If syntax error occurs then message is printed by error listener
-            // and parser throws this runtime exception to stop parsing.
-            // Just stop processing current Javadoc comment.
-            ParseErrorMessage parseErrorMessage = errorListener.getErrorMessage();
+            ParseErrorMessage parseErrorMessage = null;
+
+            if (ex.getCause() instanceof FailedPredicateException
+                    || ex.getCause() instanceof NoViableAltException) {
+                final RecognitionException recognitionEx = (RecognitionException) ex.getCause();
+                if (recognitionEx.getCtx() instanceof JavadocParser.HtmlTagContext) {
```

Comment:
please make method `isMissedHtmlTag` and make in its javadoc explain why we care about HtmlTagContext only.
All other context classes related to html
```
 $ grep --no-filename -R "extends ParserRuleContext" * | sed -E "s/.*public static //" | sed -E "s/extends .*//" | grep -E "Tag|Html"
class HtmlElementContext 
class HtmlElementOpenContext 
class HtmlElementCloseContext 
class HtmlTagContext 
class PTagOpenContext 
class PTagCloseContext 
class LiTagOpenContext 
class LiTagCloseContext 
class TrTagOpenContext 
class TrTagCloseContext 
class TdTagOpenContext 
class TdTagCloseContext 
class ThTagOpenContext 
class ThTagCloseContext 
class BodyTagOpenContext 
class BodyTagCloseContext 
class ColgroupTagOpenContext 
class ColgroupTagCloseContext 
class DdTagOpenContext 
class DdTagCloseContext 
class DtTagOpenContext 
class DtTagCloseContext 
class HeadTagOpenContext 
class HeadTagCloseContext 
class HtmlTagOpenContext 
class HtmlTagCloseContext 
class HtmlContext 
class OptionTagOpenContext 
class OptionTagCloseContext 
class TbodyTagOpenContext 
class TbodyTagCloseContext 
class TfootTagOpenContext 
class TfootTagCloseContext 
class TheadTagOpenContext 
class TheadTagCloseContext 
class SingletonTagContext 
class AreaTagContext 
class BaseTagContext 
class BasefontTagContext 
class BrTagContext 
class ColTagContext 
class FrameTagContext 
class HrTagContext 
class ImgTagContext 
class InputTagContext 
class IsindexTagContext 
class LinkTagContext 
class MetaTagContext 
class ParamTagContext 
class WrongSinletonTagContext 
class SingletonTagNameContext 
class JavadocTagContext 
class JavadocInlineTagContext 
class HtmlCommentContext
```